src/modules/battery_status/analog_battery.cpp

```diff
@@ -22,10 +22,6 @@ AnalogBattery::AnalogBattery(int index, ModuleParams *parent) :
 {
 	char param_name[17];
 
-	_analog_param_handles.cnt_v_volt = param_find("BAT_CNT_V_VOLT");
-
-	_analog_param_handles.cnt_v_curr = param_find("BAT_CNT_V_CURR");
```

Comment:
Can you go through the airframe configs (`posix-configs/` and `ROMFS/px4fmu_common/init.d/airframes`), and remove the parameters and change the voltage dividers accordingly to get to the same result?

Reply:
It will be the same result if ADC driver provides correct v_ref. I'm not sure how those annoying macros is formed. It should work, but we need test each device.

Reply:
"ADC count to voltage" actually is LSB of ADC.

Reply:
Instead we should change `BAT_V_DIV` and `BAT_A_PER_V`

Reply:
Or we need to fix the correct v_ref reported by `drivers/adc`

Reply:
I suggest that adding extra macros which define their ADC v_ref in each `board_config.h`, or creating a new low level API in `drv_adc.h`, or developing new ADC drivers for each device :( .
There is no much boards depend on `BAT_CNT_V_*`.